sway/input/tablet.c

```diff
@@ -0,0 +1,345 @@
+#define _POSIX_C_SOURCE 200809L
+#include <stdlib.h>
+#include <wlr/backend/libinput.h>
+#include <wlr/types/wlr_tablet_v2.h>
+#include "log.h"
+#include "sway/input/cursor.h"
+#include "sway/input/seat.h"
+#include "sway/input/tablet.h"
+
+static void handle_pad_tablet_destroy(struct wl_listener *listener, void *data) {
+	struct sway_tablet_pad *pad =
+		wl_container_of(listener, pad, tablet_destroy);
+
+	pad->tablet = NULL;
+
+	wl_list_remove(&pad->tablet_destroy.link);
+	wl_list_init(&pad->tablet_destroy.link);
+}
+
+static void attach_tablet_pad(struct sway_tablet_pad *tablet_pad,
+		struct sway_tablet *tablet) {
+	sway_log(SWAY_DEBUG, "Attaching tablet pad \"%s\" to tablet tool \"%s\"",
+		tablet_pad->seat_device->input_device->wlr_device->name,
+		tablet->seat_device->input_device->wlr_device->name);
+
+	tablet_pad->tablet = tablet;
+
+	wl_list_remove(&tablet_pad->tablet_destroy.link);
+	tablet_pad->tablet_destroy.notify = handle_pad_tablet_destroy;
+	wl_signal_add(&tablet->seat_device->input_device->wlr_device->events.destroy,
+		&tablet_pad->tablet_destroy);
+}
+
+struct sway_tablet *sway_tablet_create(struct sway_seat *seat,
+		struct sway_seat_device *device) {
+	struct sway_tablet *tablet =
+		calloc(1, sizeof(struct sway_tablet));
+	if (!sway_assert(tablet, "could not allocate sway tablet for seat")) {
+		return NULL;
+	}
+
+	wl_list_insert(&seat->cursor->tablets, &tablet->link);
+
+	device->tablet = tablet;
+	tablet->seat_device = device;
+
+  return tablet;
+}
+
+void sway_configure_tablet(struct sway_tablet *tablet) {
+	struct wlr_input_device *device =
+		tablet->seat_device->input_device->wlr_device;
+  struct sway_seat *seat = tablet->seat_device->sway_seat;
```

Comment:
throughout this file really

Reply:
Hopefully fixed now. I'm not sure what kind of snafu occurred in my editor to cause this...